pkg/imgpkg/signature/cosign.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package signature
+
+import (
+	"fmt"
+	"net/http"
+
+	regname "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/k14s/imgpkg/pkg/imgpkg/imageset"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+)
+
+func NewCosignFinder(reg registry.Registry) *Cosign {
+	return &Cosign{registry: reg}
+}
+
+type Cosign struct {
+	registry registry.Registry
+}
+
+func (c Cosign) Signature(reference regname.Digest) (imageset.UnprocessedImageRef, error) {
+	tag, err := c.signatureTag(reference)
+	if err != nil {
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signature, err := c.registry.Digest(tag)
+	if err != nil {
+		if transportErr, ok := err.(*transport.Error); ok {
+			if transportErr.StatusCode == http.StatusNotFound || transportErr.StatusCode == http.StatusForbidden || transportErr.StatusCode == http.StatusUnauthorized {
+				return imageset.UnprocessedImageRef{}, NotFound{}
+			}
+		}
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signDigest, err := regname.NewDigest(reference.Name())
+	if err != nil {
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signDigest = signDigest.Digest(signature.String())
+	return imageset.UnprocessedImageRef{DigestRef: signDigest.Name(), Tag: tag.TagStr()}, nil
+}
+
+// TODO: Maybe this functions should come from cosign, nevertheless the only version
+//       they currently have is conversion from a v1.Descriptor into the tag
+func (c Cosign) signatureTag(reference regname.Digest) (regname.Tag, error) {
+	digest, err := v1.NewHash(reference.DigestStr())
+	if err != nil {
+		return regname.Tag{}, fmt.Errorf("calculating the hash: %s", err)
+	}
+
+	tag, err := regname.NewTag(reference.Repository.Name())
+	if err != nil {
+		return regname.Tag{}, fmt.Errorf("calculating the tag: %s", err)
+	}
+
+	return tag.Tag(fmt.Sprintf("%s-%s.sig", digest.Algorithm, digest.Hex)), nil
```

Comment:
decided to copy the code because the amount of libraries that cosign was adding to our code did not made sense since we are just using 1 two-line function from their code